### PR TITLE
Feature/source avatar

### LINKED ISSUE #55: Add "Avatar" column to OSINTSource model, and update frontend to show its content
Add new Column to OSINTSource to store blob of image to view in frontend StoryCard

If not set by the user a job could be triggerd to gather this: 
https://developer.apple.com/library/archive/documentation/AppleApplications/Reference/SafariWebContent/ConfiguringWebApplications/ConfiguringWebApplications.html

### PR DESCRIPTION
**Pull request by ewoiset**
closes #55 

## Summary by CodeRabbit

- **New Features**
	- Implemented avatar selection and upload for configuration settings, with specific rules for file type and size.
	- Added icon display functionality for OSINT sources, enhancing visual representation.
- **Enhancements**
	- Improved the display of source and author information in the UI, including adjustments to image rendering and component arrangement.
	- Enhanced story metadata presentation with conditional logic for a more dynamic user experience.
- **Bug Fixes**
	- Fixed issues related to the handling and display of base64-encoded icons in various components.

### CODE REVIEW

**Comment by ewoiset** [comment]
@b3n4kh not quite sure why loadStories() -> assessStore.updateStoryByID(story_id) is called onMounted in StoryView... I just added updateOSINTSources and left it like this... maybe there is a cleaner version though

**Comment by b3n4kh** [comment]
> @b3n4kh not quite sure why loadStories() -> assessStore.updateStoryByID(story_id) is called onMounted in StoryView... I just added updateOSINTSources and left it like this... maybe there is a cleaner version though

This loads the Story if it is not yet in the store.

**Review by b3n4kh — CHANGES_REQUESTED** [review]

File: src/gui/src/components/config/EditConfig.vue [hunk 1]
```diff
@@ -153,6 +162,23 @@ export default {
       default: null
     }
   },
+  data: () => ({
```
[the remark this answers is not in the document]
Comment (ewoiset) [reply src/gui/src/components/config/EditConfig.vue:165]: This is not correct. I have tried only using accept - however, this only preselects the image types in the upload browser, but anything can be selected and uploaded. therefore i added the rules

File: src/core/run.py
[the remark this answers is not in the document]
Comment (ewoiset) [reply src/core/run.py:1]: This seems to have happened when i merged origin/master on march 4th.. is it still like this on master?

File: src/gui/src/components/config/EditConfig.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (ewoiset) [reply src/gui/src/components/config/EditConfig.vue:165]: i will have a look into the composition API style

[the remark this answers is not in the document]
Comment (b3n4kh) [reply src/gui/src/components/config/EditConfig.vue:165]: This is certainly not the expected behavior the list of selectable files should be filtered, see screenshot below.

There are however some browser which do still not fully support this: https://caniuse.com/input-file-accept, I would however argue supporting them is out of scope, and would still, go with a non-redundant approach. If there is still a reason for an exception to this, I would add a comment why, on which devices, this additional check was necessary.

image

**Pull request merged**